cmd/space/generate.go

```diff
@@ -0,0 +1,181 @@
+				return cmdutil.NewHandledCliError(fmt.Errorf("account %s is not logged in", grantAccount))
+			}
+		} else {
+			switch {
+			case len(accounts) == 0:
+				fmt.Printf("No accounts are logged in yet. Use `guppy login <account>` to log in.")
+				return cmdutil.NewHandledCliError(fmt.Errorf("account %s is not logged in", grantAccount))
+			case len(accounts) == 1:
+				grantAccount = accounts[0]
+			default:
+				fmt.Printf("Multiple accounts are logged in. Specify an account with `--grant-to`.")
+				return cmdutil.NewHandledCliError(fmt.Errorf("multiple accounts are logged in"))
+			}
+		}
+
+		if provisionAccount == did.Undef {
+			return fmt.Errorf("no account found to provision space to")
+		}
+		if grantAccount == did.Undef {
+			return fmt.Errorf("no account found to grant space access to")
+		}
+
+		fmt.Printf("Provisioning %s to %s...\n", space.DID(), provisionAccount)
+		_, err = c.ProviderAdd(cmd.Context(), provisionAccount, c.Connection().ID().DID(), space.DID())
+		if err != nil {
+			return fmt.Errorf("provisioning space: %w", err)
+		}
+
+		fmt.Printf("Granting access on %s to %s...\n", space.DID(), grantAccount)
+
+		// Build the capabilities to grant
+		capabilities := make([]ucan.Capability[ucan.NoCaveats], 0, len(spaceAccess))
+		for _, cap := range spaceAccess {
+			capabilities = append(capabilities, ucan.NewCapability(
+				cap,
+				space.DID().String(),
+				ucan.NoCaveats{},
+			))
+		}
+
+		_, err = grant(cmd.Context(), c, space, grantAccount, capabilities)
+		if err != nil {
+			return fmt.Errorf("granting capabilities: %w", err)
+		}
+
+		fmt.Printf("Generated space: %s\n", space.DID())
+
+		return nil
+	},
+}
+
+func init() {
+	SpaceCmd.AddCommand(generateCmd)
+	generateCmd.Flags().StringVar(&generateFlags.grantTo, "grant-to", "", "Account DID to grant space access to. Must be logged in already. (optional when exactly one account is logged in)")
+	generateCmd.Flags().StringVar(&generateFlags.provisionTo, "provision-to", "", "Account DID to provision space to. Must be logged in already. (optional when exactly one account is logged in)")
+}
+
+func grant(ctx context.Context, c *client.Client, spaceSigner principal.Signer, account did.DID, capabilities []ucan.Capability[ucan.NoCaveats]) (delegation.Delegation, error) {
+	// Create the delegation from space to account
+	delToStore, err := delegation.Delegate(
+		spaceSigner,
+		account,
+		capabilities,
+		delegation.WithNoExpiration(),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating delegation: %w", err)
+	}
+
+	delToKeep, err := delegation.Delegate(
+		spaceSigner,
+		c.Issuer().DID(),
+		capabilities,
+		delegation.WithNoExpiration(),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating delegation: %w", err)
+	}
+
+	c.AddProofs(delToStore, delToKeep)
+
+	// Store the delegation via access/delegate
+	_, err = c.AccessDelegate(ctx, spaceSigner.DID(), delToStore)
+	if err != nil {
+		return nil, fmt.Errorf("storing delegation: %w", err)
+	}
+
```

Comment:
Nit for hot storage we need to grant access to the gateway to serve... tomorrow problem though